### ISSUE TITLE
Stack trace when formatting code with certain analysis errors

### ISSUE THREAD

**Issue by natebosch** [dart-lang/dart_style#697]
Here's a minimal repro:

```dart
Future doStuff() async {
  for (final foo in bar) {
  }
  }
  return foo.bad(arg).follow();
}
```

Attempting to format this file gives:

```
Unhandled exception:
type 'SubListIterable<AnalysisError>' is not a subtype of type 'List<AnalysisError>'
#0      FormatterException.message (package:dart_style/src/exceptions.dart:26:50)
#1      processFile (package:dart_style/src/io.dart:83:24)
#2      formatPaths (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/third_party/pkg_tested/dart_style/bin/format.dart:229:12)
#3      main (file:///b/build/slave/dart-sdk-linux-dev/build/sdk/third_party/pkg_tested/dart_style/bin/format.dart:159:5)
#4      _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:277:32)
#5      _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:165:12)
```

Dart version: `Dart VM version: 2.0.0-dev.61.0 (Mon Jun 11 15:46:09 2018 +0200) on "linux_x64"`
dartfmt version: `1.0.14`

Varying some of the details gives actionable output of `Could not format because the source could not be parsed:` with specific parse errors.

**Comment by munificent** [comment]
Thanks! Yeah, looks like a bug in the syntax error reporting code.

**Comment by davidmorgan** [comment]
This is fixed in #702.
